package com.anthropic.models.messages

import com.anthropic.core.jsonMapper
import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class BrowserNewTabConfigTest {

    @Test
    fun create() {
        val browserNewTabConfig =
            BrowserNewTabConfig.builder().deferLoading(true).enabled(true).build()

        assertThat(browserNewTabConfig.deferLoading()).contains(true)
        assertThat(browserNewTabConfig.enabled()).contains(true)
    }

    @Test
    fun roundtrip() {
        val jsonMapper = jsonMapper()
        val browserNewTabConfig =
            BrowserNewTabConfig.builder().deferLoading(true).enabled(true).build()

        val roundtrippedBrowserNewTabConfig =
            jsonMapper.readValue(
                jsonMapper.writeValueAsString(browserNewTabConfig),
                jacksonTypeRef<BrowserNewTabConfig>(),
            )

        assertThat(roundtrippedBrowserNewTabConfig).isEqualTo(browserNewTabConfig)
    }
}
